include/arch/arm/arch/kernel/boot.h

```diff
@@ -12,12 +13,15 @@ cap_t create_unmapped_it_frame_cap(pptr_t pptr, bool_t use_large);
 cap_t create_mapped_it_frame_cap(cap_t pd_cap, pptr_t pptr, vptr_t vptr, asid_t asid, bool_t use_large,
                                  bool_t executable);
 
+/* This is called from assembly code and thus there are no specific types in
+ * the signature.
+ */
 void init_kernel(
-    paddr_t ui_p_reg_start,
-    paddr_t ui_p_reg_end,
-    sword_t pv_offset,
-    vptr_t  v_entry,
-    paddr_t dtb_addr_p,
-    uint32_t dtb_size
+    word_t ui_p_reg_start,
```

Comment:
I understand changing the types for dtb_size and pv_offset, but the other parameters don't need to change.  It's still valid to use types that are aliases of word_t.

Reply:
Die idae is, that at this point we are called from a raw assembly interface, so all parameters are register-sized and thus word_t is used. This function will then to the first level of semantics/casting to put the values into a specific domain. This allows the compiler to emit a warning if there are issue with the cast and we do not have to trust on implicit assumption that one type is an alias of another type.

Reply:
Ok. This does make more sense now.